### ISSUE TITLE
Using Latte in read-only (pre-generated) mode

### ISSUE THREAD

**Issue by Majkl578** [nette/latte#55]
It't not really feasible to use Latte on read-only filesystem or in an environment where PHP files should not be generated/modified. This includes for example:
- HHVM in [RepoAuthoritative mode](https://github.com/facebook/hhvm/wiki/Performance-Tuning) - any generated files are essentially forbidden, since whole code base is compiled into bytecode (and thus could be distributed entirely as a single hhbc file without source files themselves);
- Google App Engine - file system is strictly read-only

This is something that's bugging me for a while (ping @matej21), and after Latte separation, it is even worse because cache file generation is hardcoded in Engine.

Basically what I'd expect is some way to pre-generate templates and let them be used on production without need to be (re)compiled.
This is also something that could be beneficial for standard deployment as whole checking and recompiling logic could be avoided (I'm looking at you, loadCacheFile).

**Comment by JanTvrdik** [comment]
Send PR.

**Comment by Majkl578** [comment]
:)
I'd like to discuss the best approach first. In 2.1, it was somehow separated by using nette/caching, but it was removed and hardcoded into Engine. So, what do you suggest?
- Separate caching and loading from Engine (it violates SRP anyway)?
- Somehow hack Engine to support read-only mode?
- ??? (suggest other way)

**Comment by JanTvrdik** [comment]
Well I thought that a simple method `Engine::ensureThatTemplateIsCompiled($template)` would be enough.

**Comment by mishak87** [comment]
@Majkl578 Where exactly is the problem?

I assume you are running artefact in known environment made from source code?
If so you can pre/generate all templates with the knowledge of target environments tempDirectory and hash it will produce for [each template](https://github.com/nette/latte/blob/v2.3/src/Latte/Engine.php#L223).

Last resort would be to set temp directory to simple value (`null` or `latte` etc.) and load them [before calling render](https://github.com/nette/latte/blob/v2.3/src/Latte/Engine.php#L91).
